### PR TITLE
feat: only poll popular networks

### LINKED ISSUE #29055: [Bug]: Infinite Load after upgrade to `12.9.0`
### Describe the bug

I just updated to the latest version `12.9.0` and it keeps loading forever when I log in. Before you ask, yes I cleared the cache, yes I restarted the browser, yes I checked that it's the correct MetaMask version. Previously I had no such issue.

Image

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

1. Upgrade to latest MetaMask version and try log in

### Error messages or log output

```shell

```

### Detection stage

In production (default)

### Version

12.9.0

### Build type

None

### Browser

Microsoft Edge

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by bergeron**
## **Description**

To reduce the impact of showing assets across all networks, which can be unbounded, this PR scopes the portfolio view to the 9 popular networks built into metamask.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29071?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/29055

## **Manual testing steps**

1. On popular networks:
  - The token filter should allow switching between popular networks and current network
  - The filter should work as described
  - RPC requests in the background should only hit the popular networks

2. On other networks
  - The token filter should become disabled and scoped to the current network
  - RPC requests in the background should only hit the current network

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by bergeron** [comment]
depends on https://github.com/MetaMask/metamask-extension/pull/29068

File: ui/selectors/selectors.js [hunk 1]
```diff
@@ -1086,9 +1087,32 @@ export function getIsTokenNetworkFilterEqualCurrentNetwork(state) {
...
+  // Portfolio view only enabled on featured networks
+  return Object.entries(tokenNetworkFilter || {}).reduce(
+    (acc, [chainId, value]) => {
+      if (FEATURED_NETWORK_CHAIN_IDS.includes(chainId)) {
+        acc[chainId] = value;
+      }
+      return acc;
+    },
+    {},
+  );
```
Comment (gambinish) [inline ui/selectors/selectors.js:1115]: Should we care about the PORTFOLIO_VIEW feature flag here? Seems like we probably don't need to

Comment (bergeron) [reply ui/selectors/selectors.js]: i can add that https://github.com/MetaMask/metamask-extension/pull/29071/commits/2010b0202a5fe153a75ed38b1610171d9722a7a0

**Review by darkwing — DISMISSED** [review]

**Comment by bergeron** [comment]
fixing e2e test

**Review by darkwing — APPROVED** [review]

**Review by gambinish — APPROVED** [review]

**Pull request merged**